Data model #39

stfnltnr opened the pull request:
resolves #39 

The following tables, as defined within the data model design document, are created during start up of the GateWayService:

- UserTable
- OrganizationTable
- ProcessStoreTable
- Orga_Process_Map - mapping between Organizations and Processes
- RatingTable
- ReviewTable

At the moment all tables are created within the ippr_security schema since their is no separete StoreService or schema. Only the data model is created for now.

Tested only with login of existing Users from user.csv. Seems that the changes don't cause any trouble :see_no_evil:, of course there aren"t any crud operations on the new datamodel so far :construction: .

**Please review the created data model (Tables, FKs, ...) with mySql Workbench or something similar since their is no opportunity to test it elsewhere atm**

If it is ok we can proceed with the next steps:
- modify RBACService for login and registration via db instead of cached users (Issue #40)
- provide ShopService - discussion needed how we plan to do the exactly
- extend Services that is possible to provide Rest-Endpoints to perform CRUD operations on added tables

stfnltnr committed:
user:
added following fields to UserImpl.java
- String email - will it be used for login - then it needs to be unique - atm login is handled via unique username and email property isn't unique
- String password - keep in mind not hashed atm
- Long organizationId - @ManyToOne not implemented yet OrganizationRepo missing
- Date createdAt

not added - roleId is atm handled many to many tablejoin

extended UserBuilder.java with additional fields
extended RBACMappingServiceImpl.java
modified CacheUsers.java that
added column email to user.csv

atm memoryusers are created during startup. Addional columns have no values atm but they are created during startup

stfnltnr committed:
user:
added setter/getters User interface

stfnltnr committed:
At the moment all tables are created within the ippr_security schema since their is no separete StoreService atm.

The following tables, as defined within the dataModel design document, are created during startup of the GateWay:

UserTable - extended
OrganizationTable
ProcessStoreTable
Orga_Process_Map - mapping between Organizations and Processes
RatingTable
ReviewTable

For now only the data model is created. User.csv was slightly modified to store passwords within the password column of the user table - should be helpful when we will change from CachedUsers to DbUsers.

Tested only with login of existing Users from user.csv. Seems that the changes don't cause  any trouble atm.

ToDo:
- modify RBACService for login and registration via db instead of cached users (Issue #40
- provide ShopService
- extend Services that is possible to provide Rest-Endpoints to perform CRUD operations on the new data model

stfnltnr committed:
Merge branch 'dev' into data-model-#39

merge dev into feature before review

stfnltnr committed:
made getUser and appendUserInformation within `ProcessEngineCallerImpl.java` public
to remove ErrorMsg from GateWay

stfnltnr committed:
updated CHANGELOG.md

amarbajric commented on CHANGELOG.md:
```diff
@@ -4,6 +4,20 @@ This file ONLY contains changes made by the AIM17 dev-team. This means, that thi
 of the forked repository.
 The format is partially based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
+## 2018-04-24 ([#39](https://github.com/amarbajric/EBUSA-AIM17/pull/39))
```
Please change the url `https://github.com/amarbajric/EBUSA-AIM17/pull/39` to `https://github.com/amarbajric/EBUSA-AIM17/pull/68` as you are referring to your Pull Request :)
Also change the `#39` to `#68`.

So it should look like this:
`## 2018-04-24 ([#68](https://github.com/amarbajric/EBUSA-AIM17/pull/68))`

amarbajric reviewed (changes requested):
**REVIEW**
I have generated a new ER-Diagram of the `ippr_security` table by using the mysql workbench. It can be found on the SP under `Development/Data_model/ippr_security_new.pdf`
Also, I have reviewed the tables and relations and I have only found one minor point regarding the `review` table. Everything else looks very promising to me.
- `review` table does not contain attribute for storing the `reviewer/approver`. Is this on purpose or have you forgotten about this point?

I have also found a very minor mistake in the `CHANGELOG.md` file that I have commented. Other than that, I think we can merge this branch into the `dev` branch as soon as these two points have been fixed/discussed. Thank you @loete 👍

stfnltnr committed:
ChangesReviewImpl.java - uploader and approver Column now available; changed link within CHANGELOG.md

stfnltnr commented:
Ok, missing columns was a naming mistake within `ReviewImpl.java` - column u_id is now replaced by to columns `uploader` and `approver`.
Changed link in `CHANGELOG.md` to PR #68 

Pls give it another try :crossed_fingers:

amarbajric commented:
Thank you for the fast response and committed changes!
Now everything looks fine to me and can be merged!
**APPROVED** 👍

tortmann commented:
> Thank you for the fast response and committed changes!
> Now everything looks fine to me and can be merged!
> APPROVED 👍

Also approved from my side since I didnt't find any additional "mistakes"

>If it is ok we can proceed with the next steps:
>- modify RBACService for login and registration via db instead of cached users (Issue #40)
>- provide ShopService - discussion needed how we plan to do the exactly
>- extend Services that is possible to provide Rest-Endpoints to perform CRUD operations on added tables

Next steps are also approved, where **modifying the RBACService (#40) has priority** so we can hopefully present a working login and register form with the new backend during the interim presentation on Friday, 27.04

ER-diagram of implemented data model: SharePoint **Documents\Development\Data_model\ippr_security_new.pdf**
